src/compiler/program.ts

```diff
@@ -1100,6 +1121,24 @@ namespace ts {
             };
         }
 
+        function getPrependNodes(): PrependNode[] {
+            if (!options.references) {
+                return emptyArray;
+            }
+
+            const nodes: PrependNode[] = [];
+            walkProjectReferenceGraph(host, options, (_file, proj, opts) => {
+                if (opts.prepend) {
+                    const dtsFilename = changeExtension(proj.outFile, ".d.ts");
+                    const js = host.readFile(proj.outFile) || `/* Input file ${proj.outFile} was missing */\r\n`;

```

Comment:
if the file does not exist, should we error?

Reply:
Added an error upstream